modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -191,18 +232,21 @@ namespace cv {
                     fused_layer_names.back() = last_layer;
                 }
 
-                void setMaxpool(size_t kernel, size_t pad, size_t stride)
+                void setMaxpool(int kernel, int pad, int stride)
                 {
                     cv::dnn::LayerParams maxpool_param;
                     maxpool_param.set<cv::String>("pool", "max");
                     maxpool_param.set<int>("kernel_size", kernel);
-                    maxpool_param.set<int>("pad", pad);
+                    maxpool_param.set<int>("pad_l", floor((float)pad / 2));
+                    maxpool_param.set<int>("pad_r", ceil((float)pad / 2));
+                    maxpool_param.set<int>("pad_t", floor((float)pad / 2));
+                    maxpool_param.set<int>("pad_b", ceil((float)pad / 2));
+                    maxpool_param.set<bool>("ceil_mode", false);
```

Comment:
Do we have tests for both odd and even `pad` values? May I ask you to check?

Reply:
I have checked for both even and odd padding. Should I add Maxpool.cfg having two or three maxpool with both even and odd padding so that it can cover all variations ?

Reply:
I think max pooling with kernel size 3 and padding 2 will be enough. There are some tests with that configuration from other frameworks as well.